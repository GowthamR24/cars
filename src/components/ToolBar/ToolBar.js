import React from 'react';
import { Button, Row, Col, InputGroup, FormControl } from 'react-bootstrap';

import SortDropDown from '../SortDropDown/';

const ToolBar = ({
  searchText,
  setSearchText,
  handleSort,
  setShowForm,
  disabled,
}) => {
  return (
    <Row className='py-3'>
      <Col lg={8}>
        <InputGroup className='w-50'>
          <FormControl
            placeholder='Search'
            value={searchText}
            onChange={(e) => setSearchText(e.target.value)}
            disabled={disabled}
          />
        </InputGroup>
      </Col>
      <Col lg={2} className='text-end'>
        <SortDropDown handleSort={handleSort} disabled={disabled} />
      </Col>
      <Col lg={2} className='text-end'>
        <Button
          variant='primary'
          onClick={() => {
            setShowForm(true);
          }}
        >
          Add Car
        </Button>
      </Col>
    </Row>
  );
};

export default ToolBar;
