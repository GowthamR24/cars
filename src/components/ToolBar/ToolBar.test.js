import React from 'react';
import { mount, shallow } from 'enzyme';
import ToolBar from './ToolBar';

describe('ToolBar', () => {
  let wrapper;
  let props = {
    searchText: '',
    setSearchText: jest.fn(),
    handleSort: jest.fn(),
    setShowForm: jest.fn(),
    disabled: false,
  };

  it('should render correctly', () => {
    wrapper = shallow(<ToolBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Check components functionality', () => {
    wrapper = mount(<ToolBar {...props} />);

    it('should call setSearchText on change', () => {
      const event = { target: { value: 'test' } };
      wrapper.find('FormControl').simulate('change', event);
      expect(props.setSearchText).toHaveBeenCalledWith('test');
    });

    it('should call setShowForm on click', () => {
      wrapper.find('Button').simulate('click');
      expect(props.setShowForm).toHaveBeenCalled();
    });
  });
});
