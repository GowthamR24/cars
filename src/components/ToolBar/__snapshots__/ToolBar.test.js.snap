// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolBar should render correctly 1`] = `ShallowWrapper {}`;
